packages/hardhat-vyper/test/tests.ts

```diff
@@ -44,6 +45,86 @@ describe("Vyper plugin", function () {
     });
   });
 
+  describe("vyper settings", function () {
+    describe("compilation with different settings", function () {
+      useFixtureProject("compilation-with-vyper-settings");
+      useEnvironment();
+
+      it("should compile and emit artifacts", async function () {
+        await this.env.run(TASK_COMPILE);
+
+        assertFileExists(path.join("artifacts", "contracts", "A.vy", "A.json"));
+        assertFileExists(path.join("artifacts", "contracts", "B.vy", "B.json"));
+      });
+    });
+
+    describe("compilation with wrong settings", function () {
+      useFixtureProject("compilation-with-vyper-settings-failure");
+      useEnvironment();
+
+      it("should fail the compilation, invalid settings", async function () {
+        // compiler version is set to 0.3.9, which does not support the setting 'optimize'
+        await expect(this.env.run(TASK_COMPILE)).to.be.rejectedWith(
+          Error,
+          /unrecognized arguments: --optimize/
```

Comment:
can we also test evmVersion?

Reply:
Done. 
FYI: this test is a bit of a hack. Basically is very tricky to see if the evmVersion and optimize settings are actually used during the compilation. So I use this error to check that the command called includes the evmVersion and optimize params.